lib/menu.ts

```diff
@@ -113,27 +121,62 @@ export class Menu extends Disposable implements IPopupContent {
     );
     this.onDispose(() => domDispose(this.content));
 
+    if (options.startIndex !== undefined) {
+      // Cannot check which elements are visible (have an offset height) on creation since none
+      // of the items are attached to the dom yet.
+      const elems = Array.from(this.content.children).filter(elem => elem.hasAttribute('tabIndex'));
+      this._selected.set(elems[options.startIndex]);
+    }
+
     FocusLayer.create(this, this.content);
   }
 
   private _nextIndex(): void {
-    this._selIndex.set((this._selIndex.get() + 1) % this._items.length);
+    const elem = this._selected.get();
+    const content = this.content;
+    if (elem) {
```

Comment:
Doesn't this conditional prevent the "down arrow" from working after clicking on a submenu? I think if !elem, it should start with `content.firstElementChild`. Maybe using this: `getNext = (elem) => (elem && elem.nextElementSibling) || content.firstElementChild` ?